content/How to create an Activity Indicator with SwiftUI - Pablo Malvasio.md

```diff
@@ -0,0 +1,106 @@
+
+Spining view shadow:
+
+https://gist.github.com/pMalvasio/c662b9223766795031a7173b7607e98a
+
+And rounded rectangle shadows:
+
+https://gist.github.com/pMalvasio/ffd666101cbe09637cd9cce05cc6aebf
+
+
+### Animation
+
+To create the already known spinning animation we’ll use a combination of `animate` and `rotation` modifiers and a `@State` variable in order to trigger the animation as soon as the view is loaded with the `.onAppear` modifier.
+
+So, we need to add the following code before the body of our view: `@State var isAnimating = false`
+
+Now we can add the animation depending on the value of that state. That working together with an `.onAppear` modifier will do the trick.
+
+https://gist.github.com/pMalvasio/12952b393bdb6e207104d3d7448e8ba4
+
+Done. You can now call your brand new SpinnerView() from where ever you want and see how is working. Usually you will want to call it when you detect you're waiting for some task to end (like a backend request) or any local task that needs to block the user and indicates that you are performing some action in order to continue.
+
+
+### Extra time? Cool, we can make this view customizable.
+
+In a few steps we can make our new view customizable so you can call SpinnerView() with any combination of colors, size or even rotation speed.
+First we need to create a SpinnerConfiguration `struct` to allocate all the configurable parameters to call the SpinnerView.
+
+https://gist.github.com/pMalvasio/8be5fd2ed3f23169fb19690ff55fd85d
+
+Now that we successfully extracted this values from the code we can start replacing them. 
+
+Quick note: given that we code/design all in a specific set of values regarding size, we'll calculate a multiplier using that base value to accomplish a proper resize for new values.
+This multiplier will be: 
+
+`let multiplier = configuration.width / 40`
+
+And given that we need calculate this inside the body of our view, we also will be adding a `return` before the paren view.
+
+Last, but not least, we will add the following line: `var configuration: SpinnerConfiguration = SpinnerConfiguration()`. This will allow us to call our spinner like this: `SpinnerView(configuration: ...)` or without parameter given that we're providing by default a configuration object with our initial values.
+
+
+### Calling the Spinner
+
+You can now use this on any of your views just putting all your content inside a `ZStack` and drawing the SpinnerView() below that, based on any state control you have.
+
+Something like: `if showSpinner { SpinnerView() }`
+
+Being `showSpinner` a state var on my main view that's updated depending on any blocking task I'm running
+
+
+## Summary
+
+In this tutorial, you learned how to create a small but robust SpinnerView 100% made on SwiftUI. As I already mentioned before, you can see the final code (here)[https://github.com/rootstrap/spinner-playground] on a small project with a tiny demo app.
```

Comment:
`In this tutorial, you learned how to create a small but robust SpinnerView `
Sounds a bit repetitive with the first sentence of the overview.